src/Compilers/CSharp/Test/Symbol/Symbols/Source/EventTests.cs

```diff
@@ -325,6 +327,119 @@ class C
             Assert.True(fevent.RemoveMethod.IsImplicitlyDeclared, "FieldLikeEvent RemoveAccessor");
         }
 
+        [Fact]
+        public void EventWithDynamicAttribute_DynamicAttributeIsSynthesized()
+        {
+            var source = @"
+class A
+{
+        public event System.Action<dynamic> E1;
+        public event System.Action<dynamic> E2 { add {} remove {} }
+        public System.Action<dynamic> P { get; set; }
+}";
+            Action<ModuleSymbol> validator = module =>
+            {
+                var type = module.GlobalNamespace.GetMember<NamedTypeSymbol>("A");
```

Comment:
Consider casting module to PEModuleSymbol to make sure that we are checking metadata produced by the compiler.